packages/website/src/components/editor/createProvideTwoslashInlay.ts

```diff
@@ -0,0 +1,100 @@
+// The following code is adapted from the code in microsoft/TypeScript-Website.
+// Source: https://github.com/microsoft/TypeScript-Website/blob/c8b2ea8c8fc216c163fe293650b2666c8563a67d/packages/playground/src/twoslashInlays.ts
+// License: https://github.com/microsoft/TypeScript-Website/blob/c8b2ea8c8fc216c163fe293650b2666c8563a67d/LICENSE-CODE
```

Comment:
Not sure about licensing
https://github.com/microsoft/TypeScript-Website/tree/c8b2ea8c8fc216c163fe293650b2666c8563a67d?tab=readme-ov-file#legal-notices

Reply:
+1 on think of the licensing!

This should be sufficient IMO. I'm not a lawyer so don't take this as legal advice 😄 but given that we're including a link to the original MIT license & source code, I'm happy.